bin/ch-run.c

```diff
@@ -184,6 +188,63 @@ void env_delta_append(struct env_delta **ds, enum env_action act, char *arg)
    (*ds)[i].arg = arg;
 }
 
+/* Parse each line from --set-env at ':'. Each value is appended to the new
+   environment variable. '$' indicated that the variable will get expanded
+   except if --env-no-expand option is included. */
+void env_expand(char* env_set[], int argv, bool expand, char* filename)
+{
+   int i;
+   char *name, *new_value;
+   for(i = 0; i < argv; i++) {
+      if (strlen(env_set[i]) == 0 || env_set[i][0] == '\n')
+         continue; 
+      split(&name, &new_value, env_set[i], '=');
+      env_expand_error(name, filename, i+1);
+
+      // strip leading and trailing single quotes
+      if (   strlen(new_value) >= 2
+          && (new_value)[0] == '\''
+          && (new_value)[strlen(new_value) - 1] == '\'') {
+         (new_value)[strlen(new_value) - 1] = 0;  
+         (new_value)++;                           
+      }
+   
+      char *token, *r;
+      char *new_env = "";
+      r = strdup(new_value);
```

Comment:
What is the significance of `r` as a variable name?

Reply:
That was what the copy of the value being tokenized was called in the documentation for strsep. Looking back I actually don't need this variable so its getting deleted.